apps/client/src/pages/level/utils/treeLevel.ts

```diff
@@ -0,0 +1,49 @@
+export type TreeLevel = 1 | 2 | 3 | 4 | 5;
+
+export interface TreeLevelRow {
+  level: TreeLevel;
+  name: string;
+  min: number;
+  max?: number;
+  rangeLabel: string;
+}
+
+export const TREE_LEVEL_TABLE: readonly TreeLevelRow[] = [
+  { level: 1, name: '잊힌 기록의 숲', min: 0, max: 0, rangeLabel: '0개' },
+  { level: 2, name: '햇살의 터전', min: 1, max: 2, rangeLabel: '1–2개' },
+  { level: 3, name: '기록의 오솔길', min: 3, max: 4, rangeLabel: '3–4개' },
+  { level: 4, name: '지식 나무 언덕', min: 5, max: 6, rangeLabel: '5–6개' },
+  { level: 5, name: '도토리 만개 숲', min: 7, rangeLabel: '7개 이상' },
+] as const;
```

Comment:
지금도 좋은 구조지만, 만약 level에 6이 추가된다면 `TreeLevel`에도 6을 추가해줘야 하고, `TREE_LEVEL_TABLE`에도 값을 추가해줘야 해요. 규모가 작다면 크게 문제가 될 일은 없기는 하지만, 확장성을 생각했을 때 조금 불편함이 있을 것 같아요.

그렇다면 `TREE_LEVEL_TABLE`를 `as const`로 그 값 자체 리터럴 타입을 갖게 해주셨으니 여기서 `TreeLevel` type을 뽑아내는 것은 어떨까요?

```suggestion
// TREE_LEVEL_TABLE 먼저 as const로 정의 (유일한 원본)

export const TREE_LEVEL_TABLE = [
  { level: 1, name: '잊힌 기록의 숲', min: 0, max: 0, rangeLabel: '0개' },
  { level: 2, name: '햇살의 터전', min: 1, max: 2, rangeLabel: '1–2개' },
  { level: 3, name: '기록의 오솔길', min: 3, max: 4, rangeLabel: '3–4개' },
  { level: 4, name: '지식 나무 언덕', min: 5, max: 6, rangeLabel: '5–6개' },
  { level: 5, name: '도토리 만개 숲', min: 7, rangeLabel: '7개 이상' },
] as const;

// 이렇게 row 타입도 추출하고
export type TreeLevelRow = (typeof TREE_LEVEL_TABLE)[number];

// level 타입도 추출하고!
export type TreeLevel = TreeLevelRow['level'];
```

이렇게 되면 원본 `TREE_LEVEL_TABLE`만 추가해주면 알아서 타입도 적용이 되는 거죠!

```tsx
export const TREE_LEVEL_TABLE: readonly TreeLevelRow[] = [
```
그리고 as const가 애초에 readonly의 의미를 가지기도 하고요! 어떻게 생각하시나요??